PowerShell.Common.props

```diff
@@ -0,0 +1,82 @@
+
+    <FileVersion>6.0.0.0</FileVersion>
+    <AssemblyVersion>6.0.0.0</AssemblyVersion>
+
+    <ProductVersion>6.0.0-beta.3</ProductVersion>
+    <InformationalVersion>6.0.0-beta.3</InformationalVersion>
+
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+
+    <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>../signing/visualstudiopublic.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+  </PropertyGroup>
+
+  <Target Name="GetPS6VersionFromGit"
+      BeforeTargets="BeforeBuild">
+
+    <PropertyGroup>
+      <GitExe>git</GitExe>
+    </PropertyGroup>
+
+    <Exec Command='$(GitExe) rev-parse --show-cdup' ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="GitCDUP" />
+    </Exec>
+
+    <PropertyGroup>
+      <GitInfoBaseDir Condition="'$(GitInfoBaseDir)' == ''">$(MSBuildProjectDirectory)/$(GitCDUP)/.git</GitInfoBaseDir>
+    </PropertyGroup>
+
+    <Exec Command='$(GitExe) --git-dir="$(GitInfoBaseDir)" describe --abbrev=60 --long' ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="PowerShellVersion" />
+    </Exec>
+  
+  <PropertyGroup>
+    <RegexGitVersion>^v(.+)-(\d+)-g(.+)</RegexGitVersion>
+    <PS6BuildVersion>$([System.Text.RegularExpressions.Regex]::Match($(PowerShellVersion), $(RegexGitVersion)).Groups[1].Value)</PS6BuildVersion>
+    <PS6AdditionalCommits>$([System.Text.RegularExpressions.Regex]::Match($(PowerShellVersion), $(RegexGitVersion)).Groups[2].Value)</PS6AdditionalCommits>
+    <PS6CommitSHA>$([System.Text.RegularExpressions.Regex]::Match($(PowerShellVersion), $(RegexGitVersion)).Groups[3].Value)</PS6CommitSHA>
+
+    <RegexSymVer>^(\d+).(\d+).(\d+)-(.+)</RegexSymVer>
+    <PS6MajorVersion>$([System.Text.RegularExpressions.Regex]::Match($(PS6BuildVersion), $(RegexSymVer)).Groups[1].Value)</PS6MajorVersion>
+    <PS6MinorVersion>$([System.Text.RegularExpressions.Regex]::Match($(PS6BuildVersion), $(RegexSymVer)).Groups[2].Value)</PS6MinorVersion>
+    <PS6PatchVersion>$([System.Text.RegularExpressions.Regex]::Match($(PS6BuildVersion), $(RegexSymVer)).Groups[3].Value)</PS6PatchVersion>
+    <PS6LabelVersion>$([System.Text.RegularExpressions.Regex]::Match($(PS6BuildVersion), $(RegexSymVer)).Groups[4].Value)</PS6LabelVersion>
+
+    <PS6FormattedVersion Condition="'$(PS6AdditionalCommits)' == ''">$(PS6BuildVersion) SHA: $(PS6CommitSHA)</PS6FormattedVersion>
+    <PS6FormattedVersion Condition="'$(PS6FormattedVersion)' == ''">$(PS6BuildVersion) Commits: $(PS6AdditionalCommits) SHA: $(PS6CommitSHA)</PS6FormattedVersion>
+
+    <ProductVersion>$(PS6BuildVersion)</ProductVersion>
+    <!-- PackageVersion>$(PS6BuildVersion)</PackageVersion -->
+    <InformationalVersion>$(PS6FormattedVersion)</InformationalVersion>
```

Comment:
This is what I get from `Microsoft.Win32.Registry.AccessControl.dll`:
```
PS:156> $o | fl ProductVersion, FileVersion, FileName, ProductName


ProductVersion : 4.6.25428.01 built by: dlab-DDVSOWINAGE031. Commit Hash: 644b3ba1fec90828fc31353461b7d1a4aac7ea7f
FileVersion    : 4.6.25428.01
FileName       : C:\Users\dongbow\Downloads\Microsoft.Win32.Registry.AccessControl.dll
ProductName    : Microsoft® .NET Framework
```
![image](https://user-images.githubusercontent.com/127450/27662562-1c39553e-5c13-11e7-8978-16e94e936ce0.png)
![image](https://user-images.githubusercontent.com/127450/27663417-12e3952c-5c17-11e7-9030-33016dbf8c87.png)

`AssemblyInformationalVersion` maps to `ProductVersion`, while `AssemblyFileVersion` maps to `FileVersion`, maybe we should do similar thing in powershell assemblies.